### ISSUE TITLE
ENUM is not created for UUIDs

### ISSUE THREAD

**Issue by Mortinke** [Mercateo/rest-schemagen#60]
Currently the ENUMs are generated for all fields except for fileds of type UUID.

```
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
class ExampleJson {

    @NotNull
    private final UUID id;

    @NotNull
    private final UUID appId;

    @Nullable
    @Size(min = 1, max = 35)
    private final String contactName;

    @NotNull
    @Size(min = 1, max = 35)
    private final String city;

    @NotNull
    private final CountryCode countryCode;

    @NotNull
    @Size(min = 1, max = 35)
    private final String street;

    @NotNull
    @Size(min = 1, max = 10)
    private final String zipcode;
}
```

```
    Optional<Link> getUpdateLink(ExampleData exampleData) {
        ExampleJson exampleJson = new ExampleJson(exampleData.getId(), exampleData
                .getAppId(), "contactName", "city", exampleData.getCountryCode(), "street", "zipcode");
        
        final CallContext context = Parameter.createContext();

        final Parameter.Builder<ExampleJson > exampleJsonBuilder = context.builderFor(
                ExampleJson.class);

        final Parameter<ExampleJson> updateJson = exampleJsonBuilder 
                .allowValues(exampleJson)
                .defaultValue(exampleJson).build();

        return linkFactory.forCall(Rel.UPDATE, r -> r.update(
                new IdParameterBean(exampleData.getId()
                        .toString()), updateJson), context);
    }
```

Result:
```
{
                "href": "http://localhost:8080/api/addresses/48278ad2-9ba0-42c1-9ba4-c33ba8684666",
                "schema": {
                    "type": "object",
                    "properties": {
                        "appId": {
                            "type": "string",
                            "format": "uuid",
                            "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}$"
                        },
                        "city": {
                            "type": "string",
                            "default": "city",
                            "enum": [
                                "city"
                            ],
                            "minLength": 1,
                            "maxLength": 35
                        },
                        "contactName": {
                            "type": "string",
                            "default": "",
                            "enum": [
                                ""
                            ],
                            "minLength": 1,
                            "maxLength": 35
                        },
                        "countryCode": {
                            "type": "string",
                            "default": "DE",
                            "enum": [
                                "DE"
                            ]
                        },
                        "id": {
                            "type": "string",
                            "format": "uuid",
                            "pattern": "^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}$"
                        },
                        "street": {
                            "type": "string",
                            "default": "street",
                            "enum": [
                                "street"
                            ],
                            "minLength": 1,
                            "maxLength": 35
                        },
                        "zipcode": {
                            "type": "string",
                            "default": "zipcode",
                            "enum": [
                                "zipcode"
                            ],
                            "minLength": 1,
                            "maxLength": 10
                        }
                    },
                    "required": [
                        "appId",
                        "city",
                        "companyName",
                        "countryCode",
                        "id",
                        "street",
                        "zipcode"
                    ]
                },
                "rel": "update",
                "mediaType": "application/json",
                "method": "PUT"
            }
```

**Comment by skynetsonar** [comment]
Integer fields do not allow enum values either

**Comment by skynetsonar** [comment]
Resolved in version 0.19.1

**Comment by skynetsonar** [comment]
Resolved in version 0.19.1
